subsys/bluetooth/shell/bt.c

```diff
@@ -193,65 +193,128 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
+static void print_data_hex(const uint8_t *data, uint8_t len, bool warning)
+{
+	shell_fprintf(ctx_shell, warning ? SHELL_WARNING : SHELL_INFO, "0x");
+	/* Reverse the byte order when printing as advertising data is LE
+	 * and the MSB should be first in the printed output.
+	 */
+	for (int8_t i = len - 1; i >= 0; i--) {
+		shell_fprintf(ctx_shell, warning ? SHELL_WARNING : SHELL_INFO, "%02x", data[i]);
+	}
+}
+
+static void print_data_set(uint8_t set_value_len,
 			   const uint8_t *scan_data, uint8_t scan_data_len)
 {
-	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+	uint8_t idx = 0;
 
-	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
-		      strlen(scan_response_label), "",
-		      type);
-	for (uint8_t i = 0U; i < (scan_data_len / min_value_len); i++) {
-		if (i > 0L) {
-			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
-		}
+	if (scan_data_len == 0)
+		return;
```

Comment:
```suggestion
	if (scan_data_len == 0) {
		return;
	}
```

It is also missing a check for `set_value_len > scan_data_len` which would also be invalid